opm/autodiff/FlowMain.hpp

```diff
@@ -129,6 +129,7 @@ namespace Opm
             asImpl().setupGridAndProps();
             asImpl().setupState();
             asImpl().distributeData();
+            asImpl().runDiagnostics();
```

Comment:
Since the diagnostics would be identical for each node in parallel (all have identical fluid properties) I think it might make more sense to run the diagnostics immediately after setupGridAndProps() instead.